core/src/main/scala/magnolia1/magnolia.scala

```diff
@@ -181,8 +181,33 @@ object Magnolia {
           annotation.tree
       }
 
-    def annotationsOf(symbol: Symbol): List[Tree] =
-      annotationTrees(symbol.annotations)
+    def annotationsOf(symbol: Symbol): List[Tree] = {
+      @tailrec
+      def fromBaseClassesMembers(owner: Symbol): List[Annotation] =
+        if (owner.isClass) {
+          owner.asClass.baseClasses
```

Comment:
it might be useful to filter out classes which for sure won't have any annotations, e.g. `java.lang.Object` or any traits from `scala.*`